src/Infrastructure/SshFileFetcher.php

```diff
@@ -24,32 +23,25 @@ class SshFileFetcher implements FileFetcher
      */
     public function fetch(PimConnection $connection, string $filePath, bool $withLocalCopy): string
     {
-        $sftp = $this->createSftpConnection($connection);
+        $ssh = new Ssh($connection->getHost(), $connection->getPort());
 
         $pathInfo = pathinfo($filePath);
         $fileName = $pathInfo['basename'];
 
-        $subList = $sftp->nlist($pathInfo['dirname']);
-
-        $filesMatchingName = array_filter($subList, function ($element) use ($fileName) {
-            return $element == $fileName;
-        });
+        $output = $ssh->exec(
+            sprintf('test -f %s && echo "File exists" || echo "File doesn\'t exists"', $filePath),
```

Comment:
use error code instead of string, much more reliable/clean. 
`test -f %s; echo "$?"` will return the error of the command.